utils/build.ps1

```diff
@@ -122,8 +127,23 @@ if ($null -ne $env:VSCMD_ARG_HOST_ARCH -or $null -ne $env:VSCMD_ARG_TGT_ARCH) {
 
 # Prevent elsewhere-installed swift modules from confusing our builds.
 $env:SDKROOT = ""
-$NativeProcessorArchName = $env:PROCESSOR_ARCHITEW6432
-if ($null -eq $NativeProcessorArchName) { $NativeProcessorArchName = $env:PROCESSOR_ARCHITECTURE }
+
+$BuildArchName = $env:PROCESSOR_ARCHITEW6432
+if ($null -eq $BuildArchName) { $BuildArchName = $env:PROCESSOR_ARCHITECTURE }
+
+if ($PinnedBuild -eq "") {
+  switch ($BuildArchName) {
+    "AMD64" {
+      $PinnedBuild = "https://download.swift.org/swift-5.10-branch/windows10/swift-5.10-DEVELOPMENT-SNAPSHOT-2024-01-18-a/swift-5.10-DEVELOPMENT-SNAPSHOT-2024-01-18-a-windows10.exe"
+      $PinnedSHA256 = "006266d8c2a6a9c70e21b9d161ec35c07bcbb8a452b17e145899d814d07a29e7"
+    }
+    "ARM64" {
+      $PinnedBuild = "https://github.com/thebrowsercompany/swift-build/releases/download/20231117.3/installer-arm64.exe"
```

Comment:
@shahmishal - this is the only ARM64 installer currently available that will run on ARM64. Since this script is used by developers and we are using it for ARM64 hosted builds, we do need to encode this.

Reply:
Removed the hardcoded path to the ARM64 installer